lib/rules/prefer-const.js

```diff
@@ -77,38 +88,143 @@ function getWriteReferenceIfOnce(variable) {
         }
     }
 
+    // Checks the writer is located at the same scope and can modify it as const.
```

Comment:
mere suggestion: pulling this expression into a named const could make the code more self-documenting.
```
const sameScopeAndModifiable = retv && retv.from !== variable.scope && canBecomeVariableDeclaration(retv.identifier);
if (!sameScopeAndModifiable) {
  return null;
}
```